netket/optimizer/jax/stochastic_reconfiguration.py

```diff
@@ -102,11 +100,12 @@ def _shape_for_sr(grads, jac):
     return grads, jac
 
 
+_flatten_grad_and_oks = jax.jit(_shape_for_sr)
+
+
 @jit
-def _flatten_grad_and_oks(grad, oks):
-    grad, oks = _shape_for_sr(grad, oks)
-    oks -= jnp.mean(oks, axis=0)
-    return grad, oks
+def _subtract_mean_from_oks(oks, n_samp):
+    return oks - _sum_inplace(jnp.sum(oks, axis=0) / n_samp)
 
```

Comment:
Before my change the mean of ok was calculated on each rank separately only from the samples of that rank, was that on purpose in order to avoid communication? (its only one scalar per node anyway) ?

Reply:
This was... indeed a bug. And probably on me since I had rewritten that code at some point. 
Thanks for spotting it. In any case it was executed only once so... not really an overhead.

However, related to my comment above, can you please not use the argument `n_samp` unless there's a good reason, instead just compute it when you need it?

Reply:
done. 61b0beeb81e7ff651e3b08cb75fb476ac263b5cb